### ISSUE TITLE
BUG: eigsh and eigs return wrong eigenvalues on arm64

### ISSUE THREAD

**Issue by mhavu** [scipy/scipy#16875]
### Describe your issue.

For some complex Hermitian matrices, `scipy.sparse.linalg.eigsh` and `scipy.sparse.linalg.eigs` return wrong (huge) eigenvalues on arm64. The code example includes the the smallest such matrix I have encountered so far. (It has rank 25, so the example is rather unwieldy.)

I am running macOS 12.4 on Apple M1 Pro. I have a conda environment that runs osx-arm64 code natively and another one that runs osx-64 (x86-64) code via Rosetta 2 translation. The bug manifests itself only in the osx-arm64 environment. Both environments use the following package versions from conda-forge:
- python 3.9.13
- numpy 1.23.2
- scipy 1.9.0
- libopenblas 0.3.21
- liblapack 3.9.0.

### Reproducing Code Example

```python
import numpy as np
from scipy.linalg import eigh
from scipy.sparse.linalg import eigsh, eigs

A = np.array([[3.26611092e-03+1.69284721e-20j, 2.02754743e-03+4.66747284e-05j,  2.25370971e-04-1.01782876e-04j, 6.32692504e-04+1.26922104e-04j,  2.63012702e-03-2.58031022e-05j, 2.00483728e-03+3.34892314e-05j,  1.29741999e-03+1.29482062e-04j, 1.78571035e-04+8.53322596e-05j,  3.61914760e-04-2.01062861e-04j, 1.85834218e-03-2.51205278e-04j,  2.53436885e-04-8.79398971e-05j, 3.15672188e-04-1.73540410e-05j,  -2.65589554e-04+6.67616327e-06j, 1.60861227e-04-1.28334702e-04j,  1.33500619e-04+3.53252985e-05j, 5.53603909e-04+1.44844241e-04j,  3.52163520e-04-1.81140815e-04j, 7.34238396e-05-2.39011558e-05j,  3.38814591e-04+1.49087782e-04j, 4.90647406e-04+1.39859895e-04j,  2.64924146e-03-2.61356659e-05j, 1.82288400e-03-2.53485397e-04j,  1.32111500e-04-2.81044688e-05j, 5.51493927e-04+1.31718390e-04j,  2.18679209e-03+1.65295037e-07j],  [ 2.02754743e-03-4.66747284e-05j, 8.89141218e-03-4.87228866e-20j,  1.81140171e-03-4.58603601e-05j, 5.39354138e-04+2.72142260e-05j,  6.58805279e-04-1.43278011e-04j, 1.18501623e-03-1.87064040e-06j,  5.45651986e-03+1.12773971e-04j, 1.07046695e-03+2.87364434e-04j,  2.29910986e-04-9.43869683e-05j, 5.82038813e-04-2.46735022e-04j,  1.05994627e-04-6.12355222e-05j, 8.15642222e-04-1.15585325e-04j,  -1.52372863e-04+2.10831677e-07j, 3.56780087e-04-2.09749553e-04j,  -3.43956712e-05+2.75611470e-05j, 4.43546405e-04+3.67380301e-04j,  1.78665713e-03+2.50017452e-04j, 6.16310454e-04+1.14754762e-05j,  7.34403080e-05+4.24937778e-05j, 2.73183555e-04+5.13436698e-05j,  1.78732075e-03+2.06454546e-04j, 7.69673688e-03+6.23521146e-05j,  1.67063848e-03-6.30761280e-05j, 3.59422253e-04+4.98532207e-06j,  5.47680824e-04-1.29467207e-04j],  [ 2.25370971e-04+1.01782876e-04j, 1.81140171e-03+4.58603601e-05j,  9.02474002e-03+9.92758529e-21j, 1.87430323e-03+6.01692846e-06j,  1.75560570e-04+1.74676052e-05j, 1.50234516e-04+1.22484615e-04j,  1.18471995e-03+5.95216919e-05j, 5.47548636e-03+4.31643183e-04j,  9.99743986e-04+2.98708634e-05j, 1.13423683e-04+1.15312513e-04j,  -2.35769800e-05+6.00201419e-05j, 3.23125700e-04+1.49420032e-05j,  6.38653641e-04+5.66132529e-05j, 2.98238722e-04-1.42260324e-04j,  -1.55467651e-04+3.21611035e-05j, 9.15303976e-05+6.69289533e-05j,  6.02957628e-04+1.47705179e-04j, 1.81244968e-03+4.03756865e-04j,  3.64140014e-04+1.94418311e-04j, -3.18716459e-05+6.34097240e-05j,  1.77669746e-04-1.98243887e-06j, 1.69213685e-03+1.44858614e-04j,  7.74660019e-03+3.56206088e-06j, 1.67481410e-03+7.15100844e-05j,  1.34398435e-04+2.17807978e-05j],  [ 6.32692504e-04-1.26922104e-04j, 5.39354138e-04-2.72142260e-05j,  1.87430323e-03-6.01692846e-06j, 9.11463454e-03+1.97664547e-20j,  2.08220155e-03-1.53065276e-04j, 5.00809804e-04-2.93325437e-04j,  4.16436413e-04-1.26412612e-04j, 1.24636708e-03+5.24904675e-05j,  5.44970044e-03+1.97904831e-04j, 1.34631176e-03+4.16092842e-06j,  1.41689204e-05+3.15474422e-05j, 3.37730852e-04-1.25564639e-04j,  1.65542133e-04+1.18797253e-04j, 9.80670149e-04-4.80281178e-04j,  8.52273989e-05-1.01029291e-04j, 4.91541063e-04+2.98570229e-05j,  1.81865741e-04-7.14374113e-05j, 5.16844672e-04-1.71024499e-05j,  1.92545265e-03+6.04530177e-05j, 5.01763864e-04+2.49469525e-04j,  6.79080508e-04-9.46326884e-05j, 4.53489912e-04+4.48602289e-06j,  1.69218087e-03-1.23515937e-04j, 7.70559643e-03-3.44991810e-05j,  1.83190186e-03+2.49151642e-04j],  [ 2.63012702e-03+2.58031022e-05j, 6.58805279e-04+1.43278011e-04j,  1.75560570e-04-1.74676052e-05j, 2.08220155e-03+1.53065276e-04j,  3.28660539e-03+6.32977238e-21j, 1.72792387e-03-1.66351691e-04j,  4.59454711e-04-1.16061191e-04j, 2.21128683e-04-1.05565768e-05j,  1.13097279e-03+2.76032947e-04j, 2.04993580e-03+1.87224132e-04j,  1.49752360e-04+5.27077387e-05j, 1.39367652e-04+4.78559969e-06j,  -4.60642330e-05+1.92722297e-04j, 3.69891886e-04-1.11726517e-04j,  2.00954288e-04-3.64089116e-05j, 6.79358332e-04+1.23734435e-04j,  3.17256112e-04+7.20620381e-05j, 6.32304100e-05+1.11694813e-04j,  4.99055342e-04-1.58207331e-04j, 6.18971242e-04+1.31082543e-04j,  2.36381332e-03-2.99406627e-06j, 6.72228198e-04+9.49284967e-05j,  1.74762994e-04-6.56138865e-07j, 1.78996100e-03-2.01270856e-04j,  2.65400759e-03+2.75311592e-05j],  [ 2.00483728e-03-3.34892314e-05j, 1.18501623e-03+1.87064040e-06j,  1.50234516e-04-1.22484615e-04j, 5.00809804e-04+2.93325437e-04j,  1.72792387e-03+1.66351691e-04j, 8.80815937e-03-1.25465053e-20j,  5.49870457e-03+9.07135879e-05j, 6.07312410e-04+7.37578840e-05j,  1.75277658e-03+1.65291927e-04j, 7.63008850e-03+3.61095884e-06j,  1.60381244e-03-6.04094892e-05j, 1.20507371e-03+1.31303778e-04j,  -3.40634379e-04-5.39883951e-05j, 7.44726543e-04-2.08861738e-04j,  1.57319999e-03+1.52543460e-05j, 3.45478938e-04+1.63876974e-05j,  1.35519002e-04-1.46078339e-04j, 3.88562711e-05-7.67814771e-05j,  2.54036911e-05+2.50943366e-05j, 2.37133216e-04+1.39836509e-05j,  6.15643892e-04-1.29856172e-04j, 4.96777483e-04-2.64615358e-04j,  -3.66035989e-05-5.80978842e-05j, 2.70383645e-04-2.99856832e-05j,  4.93900344e-04-1.29811014e-04j],  [ 1.29741999e-03-1.29482062e-04j, 5.45651986e-03-1.12773971e-04j,  1.18471995e-03-5.95216919e-05j, 4.16436413e-04+1.26412612e-04j,  4.59454711e-04+1.16061191e-04j, 5.49870457e-03-9.07135879e-05j,  2.45391090e-02+7.57490407e-20j, 5.10226285e-03+6.45371930e-04j,  1.36702612e-03+1.62421204e-05j, 1.91598602e-03-6.87792454e-05j,  1.06705520e-03-8.85148587e-06j, 5.40960293e-03+2.54857400e-04j,  6.36278136e-04+3.81032217e-04j, 6.55065490e-04-7.88897568e-05j,  2.55363300e-04+4.69255668e-05j, 3.74075163e-04+1.50519248e-04j,  1.33340346e-03-2.00298419e-05j, 4.05945812e-04+1.65974447e-04j,  -9.49518494e-05-3.30115654e-05j, 3.75321557e-05-1.15118885e-05j,  5.05214060e-04+1.41306482e-04j, 1.92199073e-03-4.76571620e-05j,  3.65778882e-04-2.23458778e-04j, 6.73829192e-05-6.24270732e-05j,  3.39203768e-04-1.69180968e-04j],  [ 1.78571035e-04-8.53322596e-05j, 1.07046695e-03-2.87364434e-04j,  5.47548636e-03-4.31643183e-04j, 1.24636708e-03-5.24904675e-05j,  2.21128683e-04+1.05565768e-05j, 6.07312410e-04-7.37578840e-05j,  5.10226285e-03-6.45371930e-04j, 2.46448360e-02+6.75956562e-21j,  4.94437904e-03-2.52276771e-04j, 7.65798661e-04+3.12928177e-04j,  -6.54382704e-06-1.51954873e-04j, 1.44763654e-03-4.21076107e-04j,  4.63007007e-03+3.80861537e-04j, 1.42775864e-03-2.29018194e-04j,  -2.10894401e-05+8.20846102e-05j, 8.16911566e-05-7.42056460e-05j,  2.40027919e-04-2.85592455e-04j, 1.36038317e-03+6.23879821e-06j,  3.86448275e-04-1.92125364e-04j, 6.71917000e-05+6.10459344e-05j,  6.86546370e-05-1.13698829e-04j, 5.07057025e-04+4.12889474e-05j,  1.82678547e-03-4.00982437e-04j, 6.38077538e-04-6.05810732e-05j,  8.80288466e-05-5.32976816e-06j],  [ 3.61914760e-04+2.01062861e-04j, 2.29910986e-04+9.43869683e-05j,  9.99743986e-04-2.98708634e-05j, 5.44970044e-03-1.97904831e-04j,  1.13097279e-03-2.76032947e-04j, 1.75277658e-03-1.65291927e-04j,  1.36702612e-03-1.62421204e-05j, 4.94437904e-03+2.52276771e-04j,  2.40155282e-02-4.95159755e-20j, 5.49720016e-03-6.18260439e-05j,  3.65990548e-04-3.19167591e-05j, 4.88906779e-04-1.60616508e-04j,  4.60634691e-04+5.77331860e-04j, 5.18561533e-03+9.07449100e-05j,  8.73269414e-04+8.51680563e-05j, 1.60207310e-04+8.36404309e-05j,  -1.40905820e-04-2.61511719e-05j, 2.67890155e-04+3.09782899e-04j,  1.34501050e-03+1.83005031e-05j, 1.88052495e-04+1.39356409e-04j,  3.29244236e-04-9.99387804e-05j, 1.92632922e-04+3.74268141e-05j,  6.30369656e-04-1.75057612e-04j, 1.81448278e-03-2.71983540e-04j,  3.73917955e-04+1.53362378e-04j],  [ 1.85834218e-03+2.51205278e-04j, 5.82038813e-04+2.46735022e-04j,  1.13423683e-04-1.15312513e-04j, 1.34631176e-03-4.16092842e-06j,  2.04993580e-03-1.87224132e-04j, 7.63008850e-03-3.61095884e-06j,  1.91598602e-03+6.87792454e-05j, 7.65798661e-04-3.12928177e-04j,  5.49720016e-03+6.18260439e-05j, 9.01398411e-03+4.38415706e-20j,  1.47588217e-03-6.62293506e-05j, 5.70838273e-04-3.00434770e-04j,  -4.67873681e-05+7.94260485e-05j, 1.36458675e-03-9.33649633e-05j,  1.68854565e-03+2.28156028e-05j, 3.56995550e-04-1.62478432e-06j,  1.08364166e-04-9.51844248e-05j, 5.09795640e-05+8.92218574e-05j,  3.48850505e-04-1.23809780e-04j, 3.53146457e-04+1.08376908e-05j,  6.74636105e-04-1.23758010e-04j, 4.82416935e-04-4.85163250e-05j,  8.40863365e-05-6.39955517e-05j, 4.44511539e-04-3.48038588e-04j,  5.68773427e-04-1.23235324e-04j],  [ 2.53436885e-04+8.79398971e-05j, 1.05994627e-04+6.12355222e-05j,  -2.35769800e-05-6.00201419e-05j, 1.41689204e-05-3.15474422e-05j,  1.49752360e-04-5.27077387e-05j, 1.60381244e-03+6.04094892e-05j,  1.06705520e-03+8.85148587e-06j, -6.54382704e-06+1.51954873e-04j,  3.65990548e-04+3.19167591e-05j, 1.47588217e-03+6.62293506e-05j,  8.54564011e-03+5.23135480e-21j, 5.42022323e-03+2.65530008e-04j,  1.28288552e-04-6.18743238e-05j, 1.67763588e-03+6.83184223e-05j,  7.22050362e-03-4.46837563e-06j, 1.71657491e-03-1.37640930e-05j,  8.82754966e-04-5.62721683e-05j, -1.98295847e-05-6.12209884e-05j,  2.38422180e-04-2.03261800e-05j, 1.57259312e-03-1.39465877e-05j,  1.94690328e-04+3.33501700e-05j, 8.05487970e-05+1.06212059e-04j,  -1.60530435e-04-4.13443907e-05j, -4.35342148e-05-1.77292770e-05j,  1.23924349e-04-3.61841684e-05j],  [ 3.15672188e-04+1.73540410e-05j, 8.15642222e-04+1.15585325e-04j,  3.23125700e-04-1.49420032e-05j, 3.37730852e-04+1.25564639e-04j,  1.39367652e-04-4.78559969e-06j, 1.20507371e-03-1.31303778e-04j,  5.40960293e-03-2.54857400e-04j, 1.44763654e-03+4.21076107e-04j,  4.88906779e-04+1.60616508e-04j, 5.70838273e-04+3.00434770e-04j,  5.42022323e-03-2.65530008e-04j, 2.49170666e-02+9.83892382e-20j,  4.66638300e-03+6.55305376e-04j, 1.67426933e-03-1.95139070e-06j,  1.66429053e-03-9.42387674e-05j, 1.42281433e-03+6.25156901e-05j,  5.19811447e-03-6.85607043e-05j, 1.45204912e-03+2.36982308e-04j,  6.95513671e-04+9.87369150e-05j, 7.95909073e-04+1.91038942e-04j,  3.77562203e-04+1.05125799e-04j, 9.94976778e-04+4.68358315e-04j,  3.06146648e-04+1.34791297e-04j, 3.74914252e-04+1.93297972e-04j,  1.74877973e-04+1.17203868e-04j],  [-2.65589554e-04-6.67616327e-06j, -1.52372863e-04-2.10831677e-07j,  6.38653641e-04-5.66132529e-05j, 1.65542133e-04-1.18797253e-04j,  -4.60642330e-05-1.92722297e-04j, -3.40634379e-04+5.39883951e-05j,  6.36278136e-04-3.81032217e-04j, 4.63007007e-03-3.80861537e-04j,  4.60634691e-04-5.77331860e-04j, -4.67873681e-05-7.94260485e-05j,  1.28288552e-04+6.18743238e-05j, 4.66638300e-03-6.55305376e-04j,  2.32235591e-02+1.94703288e-20j, 4.61533278e-03-6.17990734e-04j,  1.02864523e-04+6.58160605e-05j, -2.25704190e-05-9.98436159e-05j,  4.12320913e-04-5.33762362e-04j, 4.60840042e-03-3.48627795e-04j,  6.39818821e-04-3.88859344e-04j, -2.91071826e-04+2.18960343e-05j,  -3.69131965e-05-1.96786334e-04j, 1.72162936e-04-9.30468906e-05j,  6.13597882e-04-3.45159390e-05j, -1.47485532e-04+9.15730136e-06j,  -2.51367846e-04-1.18379458e-05j],  [ 1.60861227e-04+1.28334702e-04j, 3.56780087e-04+2.09749553e-04j,  2.98238722e-04+1.42260324e-04j, 9.80670149e-04+4.80281178e-04j,  3.69891886e-04+1.11726517e-04j, 7.44726543e-04+2.08861738e-04j,  6.55065490e-04+7.88897568e-05j, 1.42775864e-03+2.29018194e-04j,  5.18561533e-03-9.07449100e-05j, 1.36458675e-03+9.33649633e-05j,  1.67763588e-03-6.83184223e-05j, 1.67426933e-03+1.95139070e-06j,  4.61533278e-03+6.17990734e-04j, 2.48650531e-02-4.13867505e-20j,  5.39448302e-03-2.41922468e-04j, 5.98751375e-04+2.81531232e-04j,  4.73067042e-04+1.86098509e-04j, 1.46596780e-03+4.10606770e-04j,  5.38905548e-03-2.69869901e-04j, 1.22501628e-03-1.62262747e-04j,  1.47881320e-04+1.10219216e-06j, 3.60730988e-04+1.28525170e-04j,  3.57558708e-04-9.05233506e-06j, 8.51757147e-04+1.14407152e-04j,  3.37438735e-04+2.47795982e-05j],  [ 1.33500619e-04-3.53252985e-05j, -3.43956712e-05-2.75611470e-05j,  -1.55467651e-04-3.21611035e-05j, 8.52273989e-05+1.01029291e-04j,  2.00954288e-04+3.64089116e-05j, 1.57319999e-03-1.52543460e-05j,  2.55363300e-04-4.69255668e-05j, -2.10894401e-05-8.20846102e-05j,  8.73269414e-04-8.51680563e-05j, 1.68854565e-03-2.28156028e-05j,  7.22050362e-03+4.46837563e-06j, 1.66429053e-03+9.42387674e-05j,  1.02864523e-04-6.58160605e-05j, 5.39448302e-03+2.41922468e-04j,  8.51956217e-03-1.11727475e-20j, 1.47374613e-03+5.85063781e-05j,  3.34575345e-04+4.41914712e-05j, -2.05429629e-05+1.64028667e-04j,  1.01732830e-03+3.28796111e-05j, 1.59629542e-03+6.66943543e-05j,  1.42955194e-04-4.86195372e-05j, 8.33765549e-06-1.99298330e-05j,  -1.11000479e-05-5.52964610e-05j, 1.10797731e-04+8.53078386e-05j,  2.54535733e-04+9.68411956e-05j],  [ 5.53603909e-04-1.44844241e-04j, 4.43546405e-04-3.67380301e-04j,  9.15303976e-05-6.69289533e-05j, 4.91541063e-04-2.98570229e-05j,  6.79358332e-04-1.23734435e-04j, 3.45478938e-04-1.63876974e-05j,  3.74075163e-04-1.50519248e-04j, 8.16911566e-05+7.42056460e-05j,  1.60207310e-04-8.36404309e-05j, 3.56995550e-04+1.62478432e-06j,  1.71657491e-03+1.37640930e-05j, 1.42281433e-03-6.25156901e-05j,  -2.25704190e-05+9.98436159e-05j, 5.98751375e-04-2.81531232e-04j,  1.47374613e-03-5.85063781e-05j, 9.02194301e-03+1.05475167e-21j,  5.51470669e-03+1.31531535e-04j, 7.66737246e-04-2.67307737e-04j,  1.87997448e-03+1.06616526e-04j, 7.60451992e-03-1.53895078e-05j,  2.04518597e-03-1.75612135e-04j, 1.35177853e-03+8.49114417e-06j,  1.13366938e-04-8.73848781e-05j, 5.67018444e-04+2.61579437e-04j,  1.84106799e-03+2.48272773e-04j],  [ 3.52163520e-04+1.81140815e-04j, 1.78665713e-03-2.50017452e-04j,  6.02957628e-04-1.47705179e-04j, 1.81865741e-04+7.14374113e-05j,  3.17256112e-04-7.20620381e-05j, 1.35519002e-04+1.46078339e-04j,  1.33340346e-03+2.00298419e-05j, 2.40027919e-04+2.85592455e-04j,  -1.40905820e-04+2.61511719e-05j, 1.08364166e-04+9.51844248e-05j,  8.82754966e-04+5.62721683e-05j, 5.19811447e-03+6.85607043e-05j,  4.12320913e-04+5.33762362e-04j, 4.73067042e-04-1.86098509e-04j,  3.34575345e-04-4.41914712e-05j, 5.51470669e-03-1.31531535e-04j,  2.40555280e-02-5.00953477e-20j, 4.92474286e-03+2.38599141e-04j,  1.31736295e-03-3.13673766e-05j, 1.74903218e-03-2.28187703e-04j,  1.13866020e-03-2.82285758e-04j, 5.47707665e-03-2.06783531e-04j,  1.00935915e-03-4.61820292e-05j, 2.72396243e-04+8.83915566e-05j,  3.83330644e-04+2.02440931e-04j],  [ 7.34238396e-05+2.39011558e-05j, 6.16310454e-04-1.14754762e-05j,  1.81244968e-03-4.03756865e-04j, 5.16844672e-04+1.71024499e-05j,  6.32304100e-05-1.11694813e-04j, 3.88562711e-05+7.67814771e-05j,  4.05945812e-04-1.65974447e-04j, 1.36038317e-03-6.23879821e-06j,  2.67890155e-04-3.09782899e-04j, 5.09795640e-05-8.92218574e-05j,  -1.98295847e-05+6.12209884e-05j, 1.45204912e-03-2.36982308e-04j,  4.60840042e-03+3.48627795e-04j, 1.46596780e-03-4.10606770e-04j,  -2.05429629e-05-1.64028667e-04j, 7.66737246e-04+2.67307737e-04j,  4.92474286e-03-2.38599141e-04j, 2.46287180e-02+1.03975917e-19j,  5.07961517e-03-6.14118732e-04j, 6.13389818e-04-9.26238736e-05j,  2.16481585e-04+2.15726452e-05j, 1.22770412e-03-4.00884059e-05j,  5.47230739e-03-4.36363587e-04j, 1.10764656e-03-2.89679792e-04j,  1.99646747e-04-6.53487660e-05j],  [ 3.38814591e-04-1.49087782e-04j, 7.34403080e-05-4.24937778e-05j,  3.64140014e-04-1.94418311e-04j, 1.92545265e-03-6.04530177e-05j,  4.99055342e-04+1.58207331e-04j, 2.54036911e-05-2.50943366e-05j,  -9.49518494e-05+3.30115654e-05j, 3.86448275e-04+1.92125364e-04j,  1.34501050e-03-1.83005031e-05j, 3.48850505e-04+1.23809780e-04j,  2.38422180e-04+2.03261800e-05j, 6.95513671e-04-9.87369150e-05j,  6.39818821e-04+3.88859344e-04j, 5.38905548e-03+2.69869901e-04j,  1.01732830e-03-3.28796111e-05j, 1.87997448e-03-1.06616526e-04j,  1.31736295e-03+3.13673766e-05j, 5.07961517e-03+6.14118732e-04j,  2.45415414e-02-3.43603179e-20j, 5.48553019e-03-1.29961010e-04j,  4.44471278e-04+1.20804757e-04j, 3.95002032e-04+1.25162855e-04j,  1.16774885e-03-5.35168965e-05j, 5.46094374e-03-1.21585132e-04j,  1.29889232e-03-1.26287911e-04j],  [ 4.90647406e-04-1.39859895e-04j, 2.73183555e-04-5.13436698e-05j,  -3.18716459e-05-6.34097240e-05j, 5.01763864e-04-2.49469525e-04j,  6.18971242e-04-1.31082543e-04j, 2.37133216e-04-1.39836509e-05j,  3.75321557e-05+1.15118885e-05j, 6.71917000e-05-6.10459344e-05j,  1.88052495e-04-1.39356409e-04j, 3.53146457e-04-1.08376908e-05j,  1.57259312e-03+1.39465877e-05j, 7.95909073e-04-1.91038942e-04j,  -2.91071826e-04-2.18960343e-05j, 1.22501628e-03+1.62262747e-04j,  1.59629542e-03-6.66943543e-05j, 7.60451992e-03+1.53895078e-05j,  1.74903218e-03+2.28187703e-04j, 6.13389818e-04+9.26238736e-05j,  5.48553019e-03+1.29961010e-04j, 8.78880455e-03-3.07759495e-20j,  1.71908985e-03+1.82676571e-04j, 5.06973409e-04+3.00654466e-04j,  1.37802248e-04-9.80301184e-05j, 1.16744170e-03+1.54040720e-05j,  1.99416562e-03-3.20685019e-05j],  [ 2.64924146e-03+2.61356659e-05j, 1.78732075e-03-2.06454546e-04j,  1.77669746e-04+1.98243887e-06j, 6.79080508e-04+9.46326884e-05j,  2.36381332e-03+2.99406627e-06j, 6.15643892e-04+1.29856172e-04j,  5.05214060e-04-1.41306482e-04j, 6.86546370e-05+1.13698829e-04j,  3.29244236e-04+9.99387804e-05j, 6.74636105e-04+1.23758010e-04j,  1.94690328e-04-3.33501700e-05j, 3.77562203e-04-1.05125799e-04j,  -3.69131965e-05+1.96786334e-04j, 1.47881320e-04-1.10219216e-06j,  1.42955194e-04+4.86195372e-05j, 2.04518597e-03+1.75612135e-04j,  1.13866020e-03+2.82285758e-04j, 2.16481585e-04-2.15726452e-05j,  4.44471278e-04-1.20804757e-04j, 1.71908985e-03-1.82676571e-04j,  3.27869427e-03-1.96610889e-21j, 2.07388638e-03+1.47070559e-04j,  1.55070558e-04-2.13158200e-05j, 6.52009426e-04+1.34084764e-04j,  2.62384302e-03+2.60489943e-05j],  [ 1.82288400e-03+2.53485397e-04j, 7.69673688e-03-6.23521146e-05j,  1.69213685e-03-1.44858614e-04j, 4.53489912e-04-4.48602289e-06j,  6.72228198e-04-9.49284967e-05j, 4.96777483e-04+2.64615358e-04j,  1.92199073e-03+4.76571620e-05j, 5.07057025e-04-4.12889474e-05j,  1.92632922e-04-3.74268141e-05j, 4.82416935e-04+4.85163250e-05j,  8.05487970e-05-1.06212059e-04j, 9.94976778e-04-4.68358315e-04j,  1.72162936e-04+9.30468906e-05j, 3.60730988e-04-1.28525170e-04j,  8.33765549e-06+1.99298330e-05j, 1.35177853e-03-8.49114417e-06j,  5.47707665e-03+2.06783531e-04j, 1.22770412e-03+4.00884059e-05j,  3.95002032e-04-1.25162855e-04j, 5.06973409e-04-3.00654466e-04j,  2.07388638e-03-1.47070559e-04j, 9.09121897e-03+2.57156953e-20j,  1.84834533e-03+2.89235466e-06j, 5.15604811e-04-2.13343283e-05j,  6.29283093e-04-1.15481517e-04j],  [ 1.32111500e-04+2.81044688e-05j, 1.67063848e-03+6.30761280e-05j,  7.74660019e-03-3.56206088e-06j, 1.69218087e-03+1.23515937e-04j,  1.74762994e-04+6.56138865e-07j, -3.66035989e-05+5.80978842e-05j,  3.65778882e-04+2.23458778e-04j, 1.82678547e-03+4.00982437e-04j,  6.30369656e-04+1.75057612e-04j, 8.40863365e-05+6.39955517e-05j,  -1.60530435e-04+4.13443907e-05j, 3.06146648e-04-1.34791297e-04j,  6.13597882e-04+3.45159390e-05j, 3.57558708e-04+9.05233506e-06j,  -1.11000479e-05+5.52964610e-05j, 1.13366938e-04+8.73848781e-05j,  1.00935915e-03+4.61820292e-05j, 5.47230739e-03+4.36363587e-04j,  1.16774885e-03+5.35168965e-05j, 1.37802248e-04+9.80301184e-05j,  1.55070558e-04+2.13158200e-05j, 1.84834533e-03-2.89235466e-06j,  9.00870041e-03+2.40902767e-21j, 1.79954702e-03+3.38377204e-05j,  2.16417459e-04+1.12756928e-04j],  [ 5.51493927e-04-1.31718390e-04j, 3.59422253e-04-4.98532207e-06j,  1.67481410e-03-7.15100844e-05j, 7.70559643e-03+3.44991810e-05j,  1.78996100e-03+2.01270856e-04j, 2.70383645e-04+2.99856832e-05j,  6.73829192e-05+6.24270732e-05j, 6.38077538e-04+6.05810732e-05j,  1.81448278e-03+2.71983540e-04j, 4.44511539e-04+3.48038588e-04j,  -4.35342148e-05+1.77292770e-05j, 3.74914252e-04-1.93297972e-04j,  -1.47485532e-04-9.15730136e-06j, 8.51757147e-04-1.14407152e-04j,  1.10797731e-04-8.53078386e-05j, 5.67018444e-04-2.61579437e-04j,  2.72396243e-04-8.83915566e-05j, 1.10764656e-03+2.89679792e-04j,  5.46094374e-03+1.21585132e-04j, 1.16744170e-03-1.54040720e-05j,  6.52009426e-04-1.34084764e-04j, 5.15604811e-04+2.13343283e-05j,  1.79954702e-03-3.38377204e-05j, 8.87837948e-03-3.90895245e-20j,  2.01888486e-03-3.96476865e-05j],  [ 2.18679209e-03-1.65295037e-07j, 5.47680824e-04+1.29467207e-04j,  1.34398435e-04-2.17807978e-05j, 1.83190186e-03-2.49151642e-04j,  2.65400759e-03-2.75311592e-05j, 4.93900344e-04+1.29811014e-04j,  3.39203768e-04+1.69180968e-04j, 8.80288466e-05+5.32976816e-06j,  3.73917955e-04-1.53362378e-04j, 5.68773427e-04+1.23235324e-04j,  1.23924349e-04+3.61841684e-05j, 1.74877973e-04-1.17203868e-04j,  -2.51367846e-04+1.18379458e-05j, 3.37438735e-04-2.47795982e-05j,  2.54535733e-04-9.68411956e-05j, 1.84106799e-03-2.48272773e-04j,  3.83330644e-04-2.02440931e-04j, 1.99646747e-04+6.53487660e-05j,  1.29889232e-03+1.26287911e-04j, 1.99416562e-03+3.20685019e-05j,  2.62384302e-03-2.60489943e-05j, 6.29283093e-04+1.15481517e-04j,  2.16417459e-04-1.12756928e-04j, 2.01888486e-03+3.96476865e-05j,  3.26402132e-03-1.69684777e-20j]])

eigh(A, eigvals_only=True)[-1]
# osx-64:    0.048960381441412223
# osx-amd64: 0.04896038144141221

eigsh(A, 1, which='LA', return_eigenvectors=False)[0]
# osx-64:    0.04896038144141237
# osx-amd64: 6980172367554427.0

eigs(A, 1, which='LR', return_eigenvectors=False)[0]
# osx-64:    (0.04896038144141225+3.2181075409285097e-19j)
# osx-amd64: (4.068803753183918e+19+4.0688037513706545e+19j)
```

### Error message

```shell
There are no error messages.
```

### SciPy/NumPy/Python version information

1.9.0 1.23.2 sys.version_info(major=3, minor=9, micro=13, releaselevel='final', serial=0)

**Comment by mhavu** [comment]
Interestingly, if I add `%timeit` to `eigsh`or `eigh` on osx-arm64, I get:

> ArpackNoConvergence: ARPACK error -1: No convergence (251 iterations, 0/1 eigenvectors converged) [ARPACK error -14: ZNAUPD did not find any eigenvalues to sufficient accuracy.]

**Comment by tupui** [comment]
Hi @mhavu, thank you for reporting. I would suspect this is the known bug around libopenblas 0.3.21. Can you try using 0.3.18 instead? On my M1 I have good results doing so.

**Comment by mhavu** [comment]
I hadn't tried building from source, but did it just now. (Is there a *meta.yaml* somewhere for `conda build`?) My `scipy.__version__` is now 1.10.0.dev0+1728.6711afa, and `scipy.tests()` gives:

> =========================== short test summary info ============================
FAILED linalg/tests/test_solvers.py::test_solve_discrete_are - AssertionError: 
FAILED sparse/linalg/_isolve/tests/test_iterative.py::test_maxiter_worsening[gmres]
= 2 failed, 37146 passed, 2189 skipped, 12121 deselected, 134 xfailed, 11 xpassed, 136 warnings in 366.46s (0:06:06) =

However, `eigsh` and `eigs` give the correct eigenvalue now.

**Comment by tupui** [comment]
Great then I suppose this is really linked to libopenblas. Before closing this, the test that you see failing are about tolerances or something else?

**Comment by mhavu** [comment]
> Great then I suppose this is really linked to libopenblas.

Do you mean the tests? I built SciPy against the same libopenblas I had before.

> Before closing this, the test that you see failing are about tolerances or something else?

Yes, they seem to be about tolerances. In `test_solve_discrete_are`, the following arrays are not almost equal to 2 decimals:
```
x = array([[0.00e+00, 1.23e-20],
           [1.23e-20, 1.64e-02]])
y = array([[0., 0.],
           [0., 0.]])
```
> E   Mismatched elements: 1 / 4 (25%)
E   Max absolute difference: 0.01635742
E   Max relative difference: inf

In `test_maxiter_worsening[gmres]`, assertion `error <= tol*best_error` fails:
> best_error = 0.0220970874869165
error      = 0.14405538214520144
tol        = 5

**Comment by tupui** [comment]
Ok then all seems good. Tolerances are known to be flaky in tests. I will close this for now as the initial issue seems to be resolved and there are works in progress to use the correct libopenblas on Conda-forge. Thanks again!

**Comment by dhomeier** [comment]
Chiming in here as it is not clear to me if those test failures really are on the radar to be fixed at conda-forge/openblas-feedstock#143.
I am seeing similar failures with libopenblas 0.3.21 on macOS 12.5.1, namely identical results for `test_solve_discrete_are` (1.64e-02 error), and a `test_maxiter_worsening[gmres]` missing the tolerance, though only slightly, when I built OpenBLAS with gcc 12.1 (rather than Xcode 13.4):
```python
        best_error = 0.01622706276063635
        error      = 0.08845091625842058
        info       = 5
        maxiter    = 5
```
(note that the `tol = 7 if platform.machine() == 'aarch64' else 5` has no effect on Monterey 12.5.1, as the system python3's `platform.machine()` returns `arm64`). I suppose the second test could easily be relaxed for all macOS, but the `solve_discrete_are` failure on case 15 : darex # 12 indicates that OpenBLAS precision has worsened from 0.3.20 to 0.3.21.
